New Indexes chapter (#770)
neo4j/docs-cypher: modules/ROOT/pages/indexes/index.adoc

> :description: Information about using indexes in Neo4j.

Reviewer: ```suggestion
:description: Information about using indexes in Neo4j.
:page-toclevels: 0
```

= Indexes

An index is a copy of specified primary data in a Neo4j database, such as nodes, relationships, or properties.
The data stored in the index provides an access path to the data in the primary storage and allows users to evaluate query filters more efficiently (and, in some cases, semantically interpret query filters).
In short, much like indexes in a book, their function in a Neo4j graph database is to make data retrieval more efficient. 

Once an index has been created, it will be automatically populated and updated by the DBMS. 

> Neo4j supports two categories of indexes: xref:indexes/search-performance-indexes/overview.adoc[search-performance indexes] (including range, text, point, and token lookup indexes) and xref:indexes/semantic-indexes/overview.adoc[semantic indexes] (including full-text and vector indexes).

Reviewer: ```suggestion
Neo4j supports two categories of indexes: 

- xref:indexes/search-performance-indexes/overview.adoc[Search-performance indexes], for speeding up data retrieval based on _exact_ matches (range, text, point, and token lookup indexes) 
- xref:indexes/semantic-indexes/overview.adoc[Semantic indexes], for _approximate_ matches and to compute similarity scores between a query string and the matching data (full-text and vector indexes).
```

[[naming-rules-and-recommendations]]

> == Naming rules and best practices

Reviewer: I'm hesitant whether we need this section at all. The information provided here is also present at the top of `CREATE INDEX`, with more details. For me this is not necessary.

> The following is true for naming indexes:

Reviewer: ```suggestion
```
